app/js/event/event.filter.js

```diff
@@ -0,0 +1,56 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('upcomingStops.event')
+        .factory('eventFilter', eventFilter);
+
+    eventFilter.$inject = ['$http'];
+
+    // function eventFilter($http) {
+    //
+    //     var getEvents = function() {
+    //         return $http.get('data/mock.events.json');
+    //     };
+    //
+    //     return {
+    //         getEvents: getEvents
+    //     };
+    //
+    // };
+
+    function getDistanceFromLatLonInKm(lat1,lon1,lat2,lon2) {
+        var R = 6371; // Radius of the earth in km
+        var dLat = deg2rad(lat2-lat1); // deg2rad below
+        var dLon = deg2rad(lon2-lon1);
+        var a =
+                Math.sin(dLat/2) * Math.sin(dLat/2) +
+                Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) *
+                Math.sin(dLon/2) * Math.sin(dLon/2);
+        var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+        var d = R * c; // Distance in km
+        return d;
+    };
+
+    function deg2rad(deg) {
+        return deg * (Math.PI/180)
+    };
+
+    var userCoordinates = {
+        "latitude": "61.497431",
+        "longtitude": "23.768822"
+    };
+
+    function proximity(userCoordinates, $http) {
```

Comment:
The problem isn't that the function won't run, but that there are two variables with the same in same scope. This function would work even if you changed the line 44 to "function proximity($http) {", as the function would then just start using the userCoordinates variable from line 39. Variables with the same name or even just similar names can easily lead to bugs, as the code seems to run fine but the variable that the function uses is different than what is expected.